core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -801,15 +807,17 @@ private FileInfo getFileInfoInternal(LockedInodePath inodePath)
       }
     }
     fileInfo.setXAttr(inode.getXAttr());
-    MountTable.Resolution resolution;
-    try {
-      resolution = mMountTable.resolve(uri);
-    } catch (InvalidPathException e) {
-      throw new FileDoesNotExistException(e.getMessage(), e);
+    if (includeUfsInformation) {
+      MountTable.Resolution resolution;
+      try {
+        resolution = mMountTable.resolve(uri);
+      } catch (InvalidPathException e) {
+        throw new FileDoesNotExistException(e.getMessage(), e);
+      }
+      AlluxioURI resolvedUri = resolution.getUri();
+      fileInfo.setUfsPath(resolvedUri.toString());
+      fileInfo.setMountId(resolution.getMountId());
```

Comment:
If these fields are not set, what will they default to? Should we have some handling on the client side `URIStatus` object to handle the unset fields if they are accessed (ie. throw some exception to tell them to use the other API)?

Reply:
Would we see any difference in these fields for files created as MUST_CACHE? Would they still resolve to a UFS path for instance?

Reply:
I don't think so, but it would be good to avoid having things like mount id = 0 or ufs path = blank.